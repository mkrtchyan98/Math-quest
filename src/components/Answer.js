import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles({
  root: {
   flex:'1 1 50%'
    }
})

const  Answer = (props) => {
 const classes = useStyles();

const   handleClick = () => {
    return props.handleClick(props.value)
  }

    const { value } = props;
    return (
      <div className={classes.root}>
        <Button  variant="contained" color="secondary" onClick={handleClick}>
           {value}
        </Button>
      </div>
    )
  
}

export default Answer;