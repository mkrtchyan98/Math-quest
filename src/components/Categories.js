import React from 'react';
import {Link} from 'react-router-dom';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    justifyContent:'center',
    minHeight:'100vh',
    alignItems:'center',
    flexDirection:'column',
    backgroundColor:'black',
    '& > *': {
      margin: theme.spacing(1),
    },
    '& a': {
     textDecoration:'none',
     color:'white',
     outline:'none'
    },
    '& h1': {
    	color:'#f97300',
    	transition:'all .3s ease-in',
    	textShadow:' 0 0 #f97300',
    '&:hover':{
    	textShadow:'10px 10px black',
    	transition:'all .5s ease-out'
    }
    }
  },
}));
const  Categories = () => {
	  const classes = useStyles();
  return (
    <div className={classes.root}>
        <h1> Arithmetic  Questions </h1>
       <ButtonGroup
        orientation="vertical"
        color="primary"
        aria-label="vertical contained primary button group"
        variant="contained"
        size="large"
      >
        <Button> <Link to="/add" >Addition</Link> </Button>
        <Button> <Link to="/sub">Subtraction</Link> </Button>
        <Button> <Link to="/multiply">Multipy</Link> </Button>
      </ButtonGroup>

    </div>
  );
}

export default Categories;
