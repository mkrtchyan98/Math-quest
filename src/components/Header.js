import React from 'react';
import { Link } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
        display:'none',
  [theme.breakpoints.down('sm')]: {
 background: 'transparent',
 display:'flex'
    },
    '& a':{
        textDecoration:'none',
        color:'#f97300',
        outline:'none'
    }
    }
}))



const Header = (props) => {
       const classes = useStyles();

    return (
            <header className={classes.root}>
            <div>
                <Link className="back-btn" to={"/"}>&larr;</Link>
                <Link to={"/"}>Arithmetic Questions</Link>
            </div>
        </header>
    )
}

export default Header