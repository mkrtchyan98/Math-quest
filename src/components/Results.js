import React from 'react';
import { Link } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';


const useStyles = makeStyles(theme => ({
  root: {
"& a": {
    outline:'none',
    color:'white',
    textDecoration:'none'
    }
}
}))



const Results = (props) => {
       const classes = useStyles();

    return (
            <div className={classes.root}>
             <h2>Right Answers: {props.rightAnswers} <hr/> Wrong Answers: {props.wrongAnswers}</h2>
             <Button  variant="contained" color="secondary"><Link to="/">Return Home</Link></Button>
                    
        </div>
    )
}

export default Results