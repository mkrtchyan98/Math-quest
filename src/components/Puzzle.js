import React from 'react';
import Answer from './Answer';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    '& div.Puzzle__Answers': {
        display:'flex',
        flexDirection:'row',
        flexWrap:'wrap',
        lineHeight:'4rem',
    },
    '& div.answer':{
        flex:'1 1 50%'
    },
     '& h1.Puzzle__Statement':{
        fontSize:'40px',
        color:'white'
    },
    
    }
})


const Puzzle = (props) => {
   const classes = useStyles();
    const { data, handleClick } = props
    return (
      <div className={classes.root}>
        <h1 className="Puzzle__Statement">
          {data.questionString}
        </h1>
        <div className="Puzzle__Answers">
          {data.answers.map((answer, i) => {
            return (
              <Answer
                key={answer}
                value={answer}
                handleClick={handleClick}
              />
            )
          })}
        </div>
      </div>
    )

}

export default Puzzle;