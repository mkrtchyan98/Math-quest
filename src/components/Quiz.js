import React, { useState } from 'react';
import Puzzle from './Puzzle';
import helpers from '../Helpers/Question';
import { makeStyles } from '@material-ui/core/styles';
import Header from './Header';
import Results from './Results';

const useStyles = makeStyles(theme =>({
  root: {
    display: 'flex',
    justifyContent:'center',
    alignItems:'center',
    flexDirection:'column',
    background:'black',
    width:'400px',
    height:'400px',
    backgroundImage:'linear-gradient(to right,rgba(152,90,21,.13), rgba(198, 153, 104, 0.17))',
    backgroundSize:'15px',
  [theme.breakpoints.down('sm')]: {
  maxWidth:'800px',
  width:'100%',
  margin:'0 auto',
  height:'100%',
  '& h1.Puzzle__Statement':{
    fontSize:'30px'
  }
  },
  '& span.Results__Symbol--Correct':{
    color:'green'
  },
  '& span.Results__Symbol--Incorrect':{
    color:'red'
  },
    '& div.Results': {
        display:'flex',
        flexDirection:'row',
        justifyContent:'end',
        alignItems:'center',
        position:'relative',
        right:'2px',

    },
    '& div.PuzzleContainer':{
        background:'#1e683d',
        width:'90%',
        height:'90%',
    },
    '& h2':{
        fontSize:'16px'
    },
    '& div.PuzzleResult':{
        display:'none'
    },
    '& div.QuestionNumber':{
        float:'left',
        fontSize:'calc(10px + 2vmin)',
        position:'relative',
        left:'2px',
    }
}
}));

const  Quiz = (props) =>  {
 const classes = useStyles();
const [counter,setCounter] = useState(0);
const [level,setLevel] = useState("single");
const [data,setData] = useState(helpers.Question("single",props.operator));
const [rightAnswers,setRightAnswers] = useState(0);
const [wrongAnswers,setWrongAnswers] = useState(0);
   const   choseLevel = (value) => {
        if (value > 5 && value <= 10) {
            setLevel("double")
        } else if (value > 10 && value <= 15) {
           setLevel("triple")
        } 
    }
   const  handleClick = (val) =>{
        if (val === data.correctAnswer) {
            setRightAnswers(rightAnswers+1)
        } else {
           setWrongAnswers(wrongAnswers+1)
        }
          setCounter(counter + 1);
          setData(helpers.Question(level,props.operator));
        if(counter > 5){
                  choseLevel(counter)
              }
        }    
        return (
            <div className={classes.root}>
            <Header  />
                <div className={counter === 15 ? "PuzzleResult" :"PuzzleContainer"}>
                <div className="QuestionNumber">
                {counter} of 15
                </div>
                    <div className="Results">
                        <div className="Results__Correct">
                            <span className="Results__Symbol--Correct">&#10004;</span>
                            <span className="Results__Count">{rightAnswers}</span>
                        </div>
                        <div className="Results__Incorrect">
                            <span className="Results__Symbol--Incorrect">&#x2718;</span>
                            <span className="Results__Count">{wrongAnswers}</span>
                        </div>
                    </div>
                    <Puzzle
                        handleClick={handleClick}
                        data={data} />
                </div>
                {
                    counter === 15 &&  
                    <Results rightAnswers={rightAnswers}  wrongAnswers={wrongAnswers} />
            }
            </div>
        );
 
}

export default Quiz