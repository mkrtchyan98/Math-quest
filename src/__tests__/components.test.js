import React from 'react';
import { shallow, mount,render } from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import Categories from '../components/Categories';
import Answer from '../components/Answer';
import Header from '../components/Header';
import Puzzle from '../components/Puzzle';
import Results from '../components/Results';
import Quiz from '../components/Quiz';


function setup() {
  const props = {
    
    data: {
    questionString: "7 + 2 =  ?",
    correctAnswer: 9,
    answers: [7, 8, 9, 4],
    handleClick:jest.fn() 
  }
    
    
  }

  const enzymeWrapper = mount(<Puzzle {...props} />)

  return {
    props,
    enzymeWrapper
  }
}


describe('<Answer />',() => {
	test('renders snapshot',() => {
		const wrapper = shallow(<Answer />);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders button',() => {
		const props ={
			value:4,
			handleClick:jest.fn(),
			counter:0

		}
		const wrapper = mount(<Answer  {...props} />);
		expect(wrapper.find('button').text()).toEqual('4');
	    expect(wrapper.find('button').simulate('click'));
	    expect(props.handleClick).toBeCalled();

	})
	})


	describe('<Header />',() => {
	test('renders snapshot',() => {
		const wrapper = shallow(<Header />);
		expect(wrapper).toMatchSnapshot();
	})
})


	describe('<Puzzle />',() => {
	test('renders snapshot',() => {
		const {enzymeWrapper} = setup();
		expect(enzymeWrapper).toMatchSnapshot();
	});
	it('renders Answer init',() => {
		const {enzymeWrapper} = setup();
		expect(Answer).toHaveLength(1);
	})
})



	describe('<Results />',() => {
	test('renders snapshot',() => {
		const wrapper = shallow(<Results />);
		expect(wrapper).toMatchSnapshot();
	})
})


	describe('<Categories />',() => {
	test('renders snapshot',() => {
		const wrapper = shallow(<Categories />);
		expect(wrapper).toMatchSnapshot();
	});

	it('valid path should  redirect to NotFound',() => {
		const wrapper = mount (
			 <MemoryRouter initialEntries={[ '/adfd' ]} initialIndex={0}>
             <Categories/>
    		 </MemoryRouter>
    		 )
    expect(wrapper.find(Quiz)).toHaveLength(0);
	})
})

