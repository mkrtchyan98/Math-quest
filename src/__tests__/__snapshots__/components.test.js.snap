// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> renders snapshot 1`] = `ShallowWrapper {}`;

exports[`<Categories /> renders snapshot 1`] = `ShallowWrapper {}`;

exports[`<Header /> renders snapshot 1`] = `ShallowWrapper {}`;

exports[`<Puzzle /> renders snapshot 1`] = `ReactWrapper {}`;

exports[`<Results /> renders snapshot 1`] = `ShallowWrapper {}`;
